src/Microsoft.ML.Tokenizers/Tokenizer.cs

```diff
@@ -136,6 +137,86 @@ public int CountTokens(string text, bool considerSpecialTokens = true)
             return idsCount;
         }
 
+        /// <summary>
+        /// Find the maximum encoding capacity from beginning within the input text without surpassing the token limit.
+        /// </summary>
+        /// <param name="text">The text to encode.</param>
+        /// <param name="maxTokenCount">The maximum token count to limit the encoding capacity.</param>
+        /// <param name="considerSpecialTokens">Indicate if want to consider the special tokens during the encoding.</param>
+        /// <returns>
+        /// - The entire normalized text.
+        /// - The length of text from the beginning of the normalized which is limited by the maximum token count
+        /// - The token count can be generated using the provided length which should be smaller than the maximum token count.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">The input text is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">The maximum token count must be greater than 0.</exception>
+        /// <remarks>
+        /// If the tokenizer has a normalizer, the returned text will be the normalized text. Otherwise the returned text will be the input text.
```

Comment:
Is it ever the case that someone might have pre-normalized text and want a method that doesn't do this normalization?

Reply:
Yes, when creating the tokenizer you have the option to provide the normalizer object. If you didn't then the tokenizer will not do any normalization before processing the text.

Reply:
Just imagining that someone might want to call the normalizer once, then tell this method that they've already done the normalization and avoid double-normalization / allocation.  My understanding of the use case for this API is to do a minimal amount of work so I was just asking myself "is there anything else that I can imagine someone might not want this method to do?" and normalization was the only thing I could imagine.

Reply:
Well written normalizer will return the original text without new allocation if there is no change from the original text. But I think processing time will be counted.

Reply:
Also, users can create a copy of the tokenizer without the normalization and can be used in such scenario too.

Reply:
Right, but I was more concerned with this API doing two things where someone might want it to do just one.  It's the same feeling @stephentoub shared offline
> The normalization stuff sneaking in here still rubs me the wrong way

I don't feel like it needs to be solved now, but it may be a topic during API review.

Reply:
The normalization is part of tokenization. It is optional for some scenario but important to other scenarios. So the API is not really doing two things more than communicating the encoding results including the change in the text. Anyway, I am open to any better suggestion that can make us avoid any confusion or to get cleaner API shape.